src/components/TableComponent/index.jsx

```diff
@@ -1,166 +1,120 @@
-              paginationLabel={`${id}-pagination`}
-              pageCount={pageCount}
-              currentPage={currentPage}
-              onPageSelect={(page) => {
-                updateUrl({ page });
-                sendEnterpriseTrackEvent(enterpriseId, 'edx.ui.enterprise.admin_portal.table.paginated', {
-                  tableId: id,
-                  page,
-                });
-              }}
-            />
+  useEffect(() => {
+    // Get initial data
+    paginateTable();
+    return clearTable;
+  }, []);
+
+  const handlePageChange = (page) => {
+    updateUrl({ page });
+    sendEnterpriseTrackEvent(enterpriseId, 'edx.ui.enterprise.admin_portal.table.paginated', {
+      tableId: id,
+      page,
+    });
+    paginateTable(parseInt(page, 10));
+  };
+
+  const renderTableContent = () => (
+    <div className={className}>
+      <div className="row">
+        <div className="col">
+          <div className="table-responsive">
+            <DataTable
+              id={id}
+              className="table-sm table-striped"
+              columns={columns}
+              data={formatData(data)}
+              isSortable={tableSortable}
+              itemCount={totalCount}
+              isLoading={loading}
+              manualSortBy
+              fetchData={sortByColumn}
+            >
+              <DataTable.TableControlBar />
+              <DataTable.Table />
+              <DataTable.TableFooter>
+                <div className="col d-flex justify-content-center">
+                  <Pagination
+                    paginationLabel={`${id}-pagination`}
+                    pageCount={pageCount}
+                    currentPage={currentPage}
+                    onPageSelect={handlePageChange}
+                  />
+                </div>
+              </DataTable.TableFooter>
+            </DataTable>
           </div>
         </div>
       </div>
-    );
-  }
-
-  renderLoadingMessage() {
-    return <TableLoadingSkeleton />;
-  }
-
-  renderErrorMessage() {
-    return (
-      <StatusAlert
-        alertType="danger"
-        iconClassName="fa fa-times-circle"
-        title="Unable to load data"
-        message={`Try refreshing your screen (${this.props.error.message})`}
-      />
-    );
-  }
-
-  renderEmptyDataMessage() {
-    return (
-      <StatusAlert
-        alertType="warning"
-        iconClassName="fa fa-exclamation-circle"
-        message="There are no results."
-      />
-    );
-  }
-
-  render() {
-    const {
-      data,
-      loading,
-      error,
-    } = this.props;
-
-    return (
-      <>
-        {error && this.renderErrorMessage()}
-        {loading && !data && this.renderLoadingMessage()}
-        {!loading && !error && data && data.length === 0
-          && this.renderEmptyDataMessage()}
-        {data && data.length > 0 && this.renderTableContent()}
-      </>
-    );
-  }
-}
+    </div>
+  );
+
+  const renderLoadingMessage = () => (
+    <TableLoadingSkeleton />
```

Comment:
We'd ideally rely on `DataTable`'s `isLoading` prop rather than defining our own loading state for the table at this point.